sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ManagementChannel.java

```diff
@@ -316,10 +325,102 @@ private Message createManagementMessage(String operation, String linkName) {
         return message;
     }
 
+    /***
+     * Create a Amqp key, value map to be used to create Amqp mesage for scheduling purpose.
+     *
+     * @param messageToSchedule The message which needs to be scheduled.
+     * @param maxMessageSize The maximum size allowed on send link.
+     * @return Map of key and value in Amqp format.
+     * @throws AmqpException When payload exceeded maximum message allowed size.
+     */
+    private Map<String, Object> createScheduleMessgeAmqpValue(ServiceBusMessage messageToSchedule, int maxMessageSize) {
+
+        Message message = messageSerializer.serialize(messageToSchedule);
+
+        // The maxsize allowed logic is from ReactorSender, this logic should be kept in sync.
+        final int payloadSize = messageSerializer.getSize(message);
+        final int allocationSize =
+            Math.min(payloadSize + MAX_MESSAGING_AMQP_HEADER_SIZE_BYTES, maxMessageSize);
+        final byte[] bytes = new byte[allocationSize];
+
+        int encodedSize;
+        try {
+            encodedSize = message.encode(bytes, 0, allocationSize);
+        } catch (BufferOverflowException exception) {
+            final String errorMessage =
+                String.format(Locale.US,
+                    "Error sending. Size of the payload exceeded maximum message size: %s kb",
+                    maxMessageSize / 1024);
+            throw logger.logExceptionAsWarning(new AmqpException(false,
+                AmqpErrorCondition.LINK_PAYLOAD_SIZE_EXCEEDED, errorMessage, exception, getErrorContext()));
+        }
+        HashMap<String, Object> messageEntry = new HashMap<>();
+        messageEntry.put(MESSAGE, new Binary(bytes, 0, encodedSize));
+        messageEntry.put(MESSAGE_ID, message.getMessageId());
+
+        Collection<HashMap<String, Object>> messageList = new LinkedList<>();
+        messageList.add(messageEntry);
+
+        Map<String, Object> requestBodyMap = new HashMap<>();
+        requestBodyMap.put(MESSAGES, messageList);
+        return requestBodyMap;
+    }
+
     private AmqpErrorContext getErrorContext() {
         return new SessionErrorContext(fullyQualifiedNamespace, entityPath);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Mono<Void> cancelScheduledMessage(long sequenceNumber) {
+        return isAuthorized(CANCEL_SCHEDULED_MESSAGE_OPERATION).thenMany(createRequestResponse.flatMap(channel -> {
```

Comment:
When .thenMany versus .then? You aren't returning a stream of items, just a Mono.